hack/run-e2e.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+kubectl version
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+ARTIFACTS_PATH=${ARTIFACTS_PATH:-"${REPO_ROOT}/e2e-logs"}
+mkdir -p "${ARTIFACTS_PATH}"
+
+# Install ginkgo
+export GO111MODULE="on"
+go get github.com/onsi/ginkgo/ginkgo@v1.16.4
+ginkgo version
+
+# Pre run e2e for extra components
+echo -e "\nRun pre run e2e"
+sh "${REPO_ROOT}"/hack/pre-run-e2e.sh
+
+# Run e2e
+echo -e "\nRun e2e"
+export GOLANG_PROTOBUF_REGISTRATION_CONFLICT=warn
```

Comment:
Just a question, who will use this environment?

Reply:
There some packages conflict that cause ginkgo to not work，adding this environment  to solves the issue

Reply:
ginkgo version is v1.16.4, now it's not needed.